### ISSUE TITLE
There are some issue with UI implementation

### ISSUE THREAD

**Issue by hoangnguyen92dn** [PhanVanLinh/NBSurveyApp#54]
- [ ] Following our design on Figma, the implemented UI quite different from the design. ICYDK, you can log in with your personal account on Figma to get the resources.

| Submission | Design |
|-|-|
|<img src="https://user-images.githubusercontent.com/6950766/123737867-f37ec980-d8cd-11eb-9653-b853f6df05ae.png" width=250>| <img src="https://user-images.githubusercontent.com/6950766/122503919-73ca4280-d023-11eb-9f70-887d82391454.png" width=250>| 

- [ ] As you are using `dp` unit for `textSize` so I am curious why don't we use `sp` instead? 

https://github.com/PhanVanLinh/NBSurveyApp/blob/9fe9ff4c822698dd42e52d7eef138296ff6b0244/app/src/main/res/layout/activity_survey_detail.xml#L17

- [ ] With the survey list, by using `RecyclerView` with `PagerSnapHelper`, it is also an approach to support scrolling with page snapping, but it will take more time when you implement the navigation indicator (bullets)

https://github.com/PhanVanLinh/NBSurveyApp/blob/9fe9ff4c822698dd42e52d7eef138296ff6b0244/app/src/main/java/com/linh/androidnbsurveyapp/ui/feature/home/HomeActivity.kt#L55-L56

- [ ] Dimension `dimens.xml` file doesn't make more sense so I am not sure about the purpose of `dp_xxx`

https://github.com/PhanVanLinh/NBSurveyApp/blob/9fe9ff4c822698dd42e52d7eef138296ff6b0244/app/src/main/res/values/dimens.xml#L3-L153

**Comment by PhanVanLinh** [comment]
1) Yes, I didn't know about it, I try to get the resource but it not possible and also I see a comment in the requirement which tell that "The design don't need to same as figma) so I though I can not get the resource. 

2) If we use `sp`, so when user modify the system text size in `Setting`, the text size in application will affect, so the UI will look different. If we use `dp`, this problem won't happened (I facing this bug from many customer before so I end up with using `dp`)

3) I think you mention about solution using `ViewPager2` with `Tab`. Actually, `ViewPager2` is base on `RecyclerView` but it is not the reason. At the begin, I choose `RecyclerView` instead `ViewPager` because I think the surveys size is flexible (due to pageNumber and pageSize) and we can load more if we scroll to the right. Therefore I use the RecyclerView to do it instead of using infinity ViewPager

4) The default `dimens` file will be helpful for support multiple screen mobile. In the future, I can create a gradle script to generate dimens size to support multiple screen. (A repository which the same idea https://github.com/intuit/sdp)

**Comment by sleepylee** [comment]
> The design don't need to same as figma

I agree that we might not be able to build exactly the same (unless you signup for Figma to get the resources). But a major issue here IMO is that on the baseline design requirement, e.g your `Forgot Password` design doesn't match the position as well.

**Comment by sleepylee** [comment]
>If we use sp, so when user modify the system text size in Setting, the text size in application will affect, so the UI will look different. If we use dp, this problem won't happened (I facing this bug from many customer before so I end up with using dp)

What if we wanted the app to display the text size according to the Zoom in setting like a natural behavior? Am I understanding correctly that we should skip it because you faced an issue before and hence this is the preferred way, even if it's against Google's recommendation? 🤔 
https://developer.android.com/training/multiscreen/screendensities
![image](https://user-images.githubusercontent.com/13435717/123744348-d69bc380-d8d8-11eb-96ba-565f6f568bef.png)

**Comment by PhanVanLinh** [comment]
I need to show the error into to email/password field so I move the forgot password to below the Login button so I move the Forgot password to below Login button

**Comment by PhanVanLinh** [comment]
I just explain the benefit of using `dp` instead of `sp` in current project. Actually, for this issue, I need to confirm with whole team to decide if we want the font to scale whenever we change `Setting` font or not. For many screen, after we change the text size in Setting, we can not see the screen content well because there is not enough space

**Comment by hoangnguyen92dn** [comment]
> If we use sp, so when user modify the system text size in Setting, the text size in application will affect, so the UI will look different. If we use dp, this problem won't happened (I facing this bug from many customer before so I end up with using dp)

As @sleepylee mentioned, using `dp` for the textSize is against Google's recommendation and we could break the UX in case I want to display the content with a bigger font size but this application doesn't display as my expectation? 🤷‍♂️ 

> I think you mention about solution using ViewPager2 with Tab. Actually, ViewPager2 is base on RecyclerView but it is not the reason. At the begin, I choose RecyclerView instead ViewPager because I think the surveys size is flexible (due to pageNumber and pageSize) and we can load more if we scroll to the right. Therefore I use the RecyclerView to do it instead of using infinity ViewPager

You understand right my idea 👍  With `ViewPager`, you can handle pagination by adding `registerOnPageChangeCallback`, and logic is the same as handling with RecyclerView. 

> The default dimens file will be helpful for support multiple screen mobile. In the future, I can create a gradle script to generate dimens size to support multiple screen. (A repository which the same idea https://github.com/intuit/sdp)

Yup. It would be nice to support multiple screen sizes and have different `dimens` resources for different screensize.  Since I only see the dp name and value are the same so it makes confusion.

**Comment by PhanVanLinh** [comment]
Thank for your feedback, next time when I need to choose `dp` or `sp`  when starting project I will consider about we allow user to user Setting font or not. Maybe I'm used to in many situation that I need to refactor whole project screen from `sp` to `dp` with and it take me many time for testing with evident so currently I start project with default value `dp`.
